### PR TITLE
Graph builder improvements

### PR DESCRIPTION
**Pull request by evansiroky**
This PR does the following:

- Removes the BusRouteStreetMatcher graph builder and associated classes/vars since they no longer have any effect on the resulting graph that is built. See https://github.com/opentripplanner/OpenTripPlanner/issues/2957.
- Refactors all graph builder buildGraph calls to use GraphBuilderModuleSummaries. These summaries can be used to improve logging and reporting of various graph build modules.
- Improves the report that is written when annotations and graph build stats are enabled (the build-config.json must have the `htmlAnnotations` set to `true`).

### CODE REVIEW

File: src/test/java/org/opentripplanner/graph_builder/module/GtfsGraphBuilderModuleTest.java [hunk 1]
```diff
@@ -20,8 +20,6 @@
 
 public class GtfsGraphBuilderModuleTest {
 
-    private static final HashMap<Class<?>, Object> _extra = new HashMap<>();
-
     private GtfsModule builder;
```
Comment (binh-dam-ibigroup) [inline src/test/java/org/opentripplanner/graph_builder/module/GtfsGraphBuilderModuleTest.java]: Variable `builder` should be declared inside each test (it is assigned in each test each time).

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Only one minor change! Thanks.

**Comment by evansiroky** [comment]
Just added a few more subtask reporters and made the change @binh-dam-ibigroup suggested.

**Review by binh-dam-ibigroup — APPROVED** [review]

**Review by landonreed — CHANGES_REQUESTED** [review]
As discussed on the team call, please add the commands to kick off the build and attach data inputs for the Portland area we're using for testing this PR and future elevation improvement PRs. Thanks!

**Comment by evansiroky** [comment]
Details shared on our company chat.

**Review by landonreed — APPROVED** [review]

**Pull request merged**